### PR TITLE
Handle exceptions

### LINKED ISSUE #129: Review Code regarding catchable Exceptions
Let's do a code reivew at where we throw exceptions or where they could be thrown (see #71) and that we handle them.
- [ ] [wporg-gp-translation-events.php](https://github.com/WordPress/wporg-gp-translation-events/blob/trunk/wporg-gp-translation-events.php)
- [ ] ...

### LINKED ISSUE #130: Review Code regarding notices and warnings
We want to make sure that our code doesn't throw notices and warnings. Please review your local logfile to see if there is a notice or warning that occurred that can be fixed.

### PR DESCRIPTION
**Pull request by psrpinto**
Fixes #129 #130 

This PR handles all exceptions. I fixed some other things along the way because I was using the IDE to look for unhandled exceptions, and the IDE showed other things that could be fixed, so I took the opportunity to fix them as well.

I also tested everything and made sure there were no notices/warnings in the log, so #130 can also be considered fixed.

### CODE REVIEW

File: includes/routes/user/my-events.php [hunk 1]
```diff
@@ -38,11 +39,23 @@ public function handle(): void {
...
-		$args                 = array(
+		$user_id = get_current_user_id();
+		$events  = get_user_meta( $user_id, Translation_Events::USER_META_KEY_ATTENDING, true ) ?: array();
+		$events  = array_keys( $events );
+
+		$current_datetime_utc = '';
+		try {
+			$current_datetime_utc = ( new DateTime( 'now', new DateTimeZone( 'UTC' ) ) )->format( 'Y-m-d H:i:s' );
+		} catch ( Exception $e ) {
+			// This can't happen because the DateTime constructor takes 'now' as argument so can't fail, neither can
```
Comment (akirk) [inline includes/routes/user/my-events.php:50]: Then let's remove this try catch?

**Comment by akirk** [comment]
Thanks for this! I started #157 this morning and I think that maybe it could handle those exceptions on DateTime better, what do you think?

**Comment by psrpinto** [comment]
Yep, #157 makes a lot of sense. I'm putting this back in draft and will revisit once #157 is merged.

**Comment by psrpinto** [comment]
Closing this one since too many things have changed since it was opened, and it's probably not relevant any more after #157.

**Pull request closed without merging**